code/modules/reagents/chemistry/reagents.dm

```diff
@@ -78,6 +78,8 @@
 	var/list/metabolized_traits
 	/// A list of traits to apply while the reagent is in a mob.
 	var/list/added_traits
+	/// A list of other reagents this interacts with if they're metabolizing in a mob
+	var/list/datum/reagent/reagent_interactions
```

Comment:
If the only thing this is used for is methamphetamine, I don't think it should be a generic system

Remember reagents are fairly hot code, so adding stuff to the base time compounds a lot.

Reply:
It's better creating a global list of all reagents and their respective interactions just like the chemical reaction lookup table than specify them on each reagent respectively.

Just looked at the next line of code which initializes a typecache for this var which is far too expensive and should atleast be static if not global

Reply:
> It's better to create a global map of all reagents and their respective interactions just like the chemical reaction lookup table than to specify them on each reagent respectively.
> 
> This would allow different reagents that could share similar effects to point back to each other and reduce memory if there is such a case
> 
> Just looked at the next line of code which initializes a typecache for this var and that's far too expensive so it should be atleast static if not global

how is that implemented?

Reply:
> If the only thing this is used for is methamphetamine, I don't think it should be a generic system
> 
> Remember reagents are fairly hot code, so adding stuff to the base time compounds a lot.

could add an on_transfer check for it instead that starts triggering the interactions. the only thing it's used for right now is meth, yes, but there are other reagents scattered around with ad hoc code for them interacting with certain other reagents; i think it would make sense to modularize this so it can be better kept track of

Reply:
> how is that implemented?

Look at how the chemical reaction look up table is implement. It's too complex to explain in this comment.

Either way putting it in the reagent is too memeory intensive so it's not an option